### PR TITLE
Reword E0392 slightly

### PR DESCRIPTION
**Pull request by estebank**
Make it clearer that a type or lifetime argument not being used can be
fixed by referencing it in a struct's fields, not just using `PhathomData`.

CC #53589.

### CODE REVIEW

File: src/librustc_typeck/check/wfcheck.rs [hunk 1]
```diff
@@ -999,11 +999,16 @@ fn report_bivariance(tcx: TyCtxt<'_>, span: Span, param_name: ast::Name) {
...
     let suggested_marker_id = tcx.lang_items().phantom_data();
     // Help is available only in presence of lang items.
-    if let Some(def_id) = suggested_marker_id {
-        err.help(&format!("consider removing `{}` or using a marker such as `{}`",
-                          param_name,
-                          tcx.def_path_str(def_id)));
-    }
+    let msg = if let Some(def_id) = suggested_marker_id {
+        format!(
+            "consider removing `{}`, refering to it in a field or using a marker such as `{}`",
```
Comment (Centril) [inline src/librustc_typeck/check/wfcheck.rs]:
```suggestion
            "consider removing `{}`, refering to it in a field, or using a marker such as `{}`",
```

**Comment by matthewjasper** [comment]
r=me with Centril's comment addressed

**Comment by Centril** [comment]
Seems like some submodules were accidentally updated?

**Comment by estebank** [comment]
@centril sigh, cleaned up

File: src/librustc_typeck/check/wfcheck.rs [hunk 2]
```diff
@@ -999,11 +999,16 @@ fn report_bivariance(tcx: TyCtxt<'_>, span: Span, param_name: ast::Name) {
...
     let suggested_marker_id = tcx.lang_items().phantom_data();
     // Help is available only in presence of lang items.
-    if let Some(def_id) = suggested_marker_id {
-        err.help(&format!("consider removing `{}` or using a marker such as `{}`",
-                          param_name,
-                          tcx.def_path_str(def_id)));
-    }
+    let msg = if let Some(def_id) = suggested_marker_id {
+        format!(
+            "consider removing `{}`, refering to it in a field, or using a marker such as `{}`",
```
Comment (mathstuf) [inline src/librustc_typeck/check/wfcheck.rs]: Shouldn't this be `referring`?

Comment (mathstuf) [reply src/librustc_typeck/check/wfcheck.rs]: And so on in other locations.

Comment (Centril) [reply src/librustc_typeck/check/wfcheck.rs]: It should yes, but let's fix it in a follow-up.

**Pull request merged**